### ISSUE TITLE
ImageSharp resizing creates images with a blank bottom

### ISSUE THREAD

**Issue by Skrypt** [OrchardCMS/OrchardCore#15064]
Sometimes ImageSharp doesn't finish to process / resize an image and it produces an image in it's cache with a blank bottom part.

So far, to resolve this issue we clean up the ImageSharp cache to reinitiate the process to create the cached image.

**Comment by Piedone** [comment]
Do you have some kind of a repro? What we see on the OC homepage might be this?

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/866e6167-f1a7-42b8-a57c-c8fa4811f979)

**Comment by Skrypt** [comment]
Here, I can't repro but it would look more like this: 

![fake-is](https://github.com/OrchardCMS/OrchardCore/assets/3228637/c02b1fe5-0140-4b0d-90ca-bae689f1d505)

**Comment by Piedone** [comment]
Are you using Azure Blob Storage? Because then https://github.com/OrchardCMS/OrchardCore/issues/15016 might help.

**Comment by Skrypt** [comment]
Yeah, this website fetches the original image from an Azure Blob Storage account.

**Comment by Piedone** [comment]
Then I suggest you make a try with the prototype code I added under the issue (or the cleaner one under the PR) and see if it helps.

**Comment by sebastienros** [comment]
Is that a bug in ImageSharp or Orchard? Have you managed to explain what triggers it?

**Comment by Skrypt** [comment]
To be honest, no, I couldn't find why it was happening. Either it was failing to download entirely the image from the blob storage because it was not entirely uploaded (which I would be surprised of). Or, a network issue happened in the middle of the resize of this image by ImageSharp which is totally arbitrary.

The issue is not really there. I think you can't prevent this from happening because of ... weather ... 
The thing I would like though is to be able to find and purge this image from the ImageSharp cache easily instead of purging it entirely.
